src/applications/vaos/actions/newAppointment.js

```diff
@@ -195,14 +195,18 @@ export function updateFacilityPageData(page, uiSchema, data) {
 
 export function updateReasonForAppointmentData(page, uiSchema, data) {
   return async dispatch => {
-    let reasonAdditionalInfo = data.reasonAdditionalInfo || '';
-
-    // Max length for reason
-    const maxTextAreaLength =
+    let reasonAdditionalInfo = data.reasonAdditionalInfo;
+    let remainingCharacters =
       REASON_MAX_CHAR_DEFAULT - data.reasonForAppointment.length - 1;
-    reasonAdditionalInfo = reasonAdditionalInfo.substr(0, maxTextAreaLength);
 
-    const remainingCharacters = maxTextAreaLength - reasonAdditionalInfo.length;
+    if (reasonAdditionalInfo) {
```

Comment:
The reason for this is that JSON Schema's `required` test checks if fields are defined or not. An empty string is defined, and so it passes the check. This is kinda silly, but we need to make sure we don't default string fields to empty strings and make sure they're undefined instead.